hack/kind/main.go

```diff
@@ -0,0 +1,263 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// manifest is a kind cluster template
+// the explicit podSubnet definition can be removed as soon as https://github.com/kubernetes-sigs/kind/commit/60074a9e67ddc8d35d3468ab137358b62a4cf723
+// will be available in a released version of kind
+const manifest = `---
+kind: Cluster
+apiVersion: {{.APIVersion}}
+networking:
+  ipFamily: {{.IPFamily}}
+{{- if eq .IPFamily "ipv6" }}
+  podSubnet: "fd00:10:244::/56"
+{{- end}}
+nodes:
+  - role: control-plane
+{{- range .WorkerNames }}
+  - role: worker
+{{- end}}
+`
+
+var k kind
+
+type startArgs struct {
+	IPFamily string
+	//APIVersion   string
+	skipSetup    bool
+	imagesToLoad []string
+	nodes        int
+	nodeImage    string
+}
+
+func (args startArgs) WorkerNames() []string {
+	var names []string
+	// kind has the following naming scheme <cluster-name>-worker, <cluster-name>-worker2 etc
+	// this is not configurable and thus explains the awkward name construction here
+	for i := 0; i < args.nodes; i++ {
+		suffix := ""
+		if i > 0 {
+			suffix = strconv.Itoa(i + 1)
+		}
+		names = append(names, fmt.Sprintf("%s-worker%s", k.clusterName, suffix))
+	}
+	return names
+}
+
+func main() {
+	cmd := &cobra.Command{
+		Use:   "setup-kind",
+		Short: "Kind test tooling",
+		Long:  "Setup and tear down of kind clusters for testing",
+	}
+	cmd.PersistentFlags().StringVar(&k.clusterName, "cluster-name", "eck-e2e", "Kind cluster name")
+	cmd.PersistentFlags().StringVar(&k.kindVersion, "kind-version", "0.9.0", "Kind version to use (currently supported on CI: 0.8.1, 0.9.0)")
+	cmd.PersistentFlags().IntVarP(&k.logLevel, "verbosity", "v", 0, "Kind log verbosity")
+
+	cmd.PersistentPreRunE = func(_ *cobra.Command, _ []string) error {
+		// check that the kind executable is installed in the right version
+		binaryName := "kind-" + k.kindVersion
+		path, err := exec.LookPath(binaryName)
+		if err != nil {
+			return err
+		}
+		// populate the kind struct with version specific information
+		k.binary = path
+		k.apiVersion = "kind.sigs.k8s.io/v1alpha3"
+		if strings.HasPrefix(k.kindVersion, "0.9") {
```

Comment:
Should we do a semver compare here to check that the version is greater than 0.8? Otherwise, if we forget to update this line with the next release of Kind (say. 0.10.0), the API version would fall back to v1alpha3.

Reply:
Good point I was cutting corners here.